### DISCUSSION TITLE
Virtualized GridLayout not Virtualizing?

### DISCUSSION THREAD

**Discussion by thomasbritton** [adobe/react-spectrum#6870] [Q&A] [answered]
Hi all,

I've been following the [following story](https://github.com/adobe/react-spectrum/blob/main/packages/react-aria-components/stories/ListBox.stories.tsx#L347) to see how to go about implementing a virtulized grid layout but I do not think it is working as it is rendering all `300` items.

Unless my understanding is incorrect it should only render the items inside the set view or the visible area?

I have a [codesandbox demo](https://codesandbox.io/p/sandbox/exciting-jones-7sysc8?file=%2Fsrc%2FApp.tsx%3A55%2C31) showing the issue. The visible view (400x400) should show and render only `12` items, but if you check the source code it is actually rendering all `300` items still.

Is this correct understanding of how Virtualization should work or have I got it wrong? And also, is there anything in the approach to set this up?

Thanks

**Comment by reidbarber** [comment, marked as answer]
Looks like you just need a set `height` and `overflowY: 'auto'` on the Listbox itself: 
https://codesandbox.io/p/sandbox/late-glade-zlzlx3

**Reply by thomasbritton** [reply]
Hey @reidbarber ,

Thanks for your reply. That was simple!

Really wish there was more documentation on the more advanced use cases such as this.

**Reply by snowystinger** [reply]
More documentation will be forthcoming as these features move from their UNSTABLE prefix to a more stable release. Thank you for your patience and feedback.
